cli/flox-rust-sdk/src/data/flox_version.rs

```diff
@@ -0,0 +1,372 @@
+        assert!(version.pre_number.is_none());
+        assert_eq!(version.num_of_commits, Some(21));
+        assert_eq!(version.commit_vcs, Some('g'));
+        assert_eq!(version.commit_sha, Some("b91c3f1".to_string()));
+    }
+
+    #[test]
+    fn test_parse_version_with_wrong_git_describe_format() {
+        assert!(matches!(
+            "1.2.3-21".parse::<FloxVersion>(),
+            Err(VersionParseError::InvalidFormat),
+        ));
+    }
+
+    #[test]
+    fn test_parse_version_with_only_commit_sha() {
+        let version: FloxVersion = "1.2.3-gb91c3f1".parse().unwrap();
+        assert_eq!(version.major, 1);
+        assert_eq!(version.minor, 2);
+        assert_eq!(version.patch, 3);
+        assert!(version.pre_name.is_none());
+        assert!(version.pre_number.is_none());
+        assert_eq!(version.commit_vcs, Some('g'));
+        assert_eq!(version.commit_sha, Some("b91c3f1".to_string()));
+        assert!(version.num_of_commits.is_none());
+    }
+
+    #[test]
+    fn test_parse_version_with_pre_release_and_commits() {
+        assert!(matches!(
+            "1.2.3-rc.1-10-gb91c3f1".parse::<FloxVersion>(),
+            Err(VersionParseError::InvalidFormat),
+        ));
+    }
+
+    #[test]
+    fn test_version_ordering() {
+        // This is the order of versions from smallest to largest
+        let v1: FloxVersion = "1.2.2".parse().unwrap();
+        let v2: FloxVersion = "1.2.2-10-gb91c3f1".parse().unwrap();
+        let v3: FloxVersion = "1.2.2-11-gb91c3f1".parse().unwrap();
+        let v4: FloxVersion = "1.2.3-rc.1".parse().unwrap();
+        let v5: FloxVersion = "1.2.3-rc.2".parse().unwrap();
+        let v6: FloxVersion = "1.2.3".parse().unwrap();
+
+        assert!(v1 == v1);
+        assert!(v1 < v2);
+        assert!(v1 < v3);
+        assert!(v1 < v4);
+        assert!(v1 < v5);
+        assert!(v1 < v6);
+
+        assert!(v2 > v1);
+        assert!(v2 == v2);
+        assert!(v2 < v3);
+        assert!(v2 < v4);
+        assert!(v2 < v5);
+        assert!(v2 < v6);
+
+        assert!(v3 > v1);
+        assert!(v3 > v2);
+        assert!(v3 == v3);
+        assert!(v3 < v4);
+        assert!(v3 < v5);
+        assert!(v3 < v6);
+
+        assert!(v4 > v1);
+        assert!(v4 > v2);
+        assert!(v4 > v3);
+        assert!(v4 == v4);
+        assert!(v4 < v5);
+        assert!(v4 < v6);
+
+        assert!(v5 > v1);
+        assert!(v5 > v2);
+        assert!(v5 > v3);
+        assert!(v5 > v4);
+        assert!(v5 == v5);
+        assert!(v5 < v6);
+
+        assert!(v6 > v1);
+        assert!(v6 > v2);
+        assert!(v6 > v3);
+        assert!(v6 > v4);
+        assert!(v6 > v5);
+        assert!(v6 == v6);
+    }
+
+    #[test]
+    fn test_version_ordering_with_flake_style_version() {
+        let v1: FloxVersion = "1.2.2".parse().unwrap();
+        let v2: FloxVersion = "1.2.2-gb91c3f1".parse().unwrap();
+
+        assert!(v1 != v2);
+        assert_eq!(v1.partial_cmp(&v2), None);
+    }
+}
```

Comment:
nonblocking suggestion: might be good to add a few cases testing roundtrips where you parse a version and then call to_string and make sure you have what you started with